src/codegen_call_json.cpp

```
@@ -10,201 +19,204 @@ static bool lookupJsonSet(const std::unordered_set<llvm::Value*> &set, llvm::Val
-    requireArgs(e, 1);
-    llvm::Value *val = emitExpr(*e.args[0]);
-    if (!lookupJsonSet(resource_sets_[RK_JsonValue], val))
-        codegenError("json_free() requires a JsonValue argument");
-    if (lookupJsonSet(json_type_only_, val))
-        codegenError("json_free() cannot free borrowed child values from get()/at()");
-    return emitResourceFree(val, RK_JsonValue, *e.args[0]);
+static llvm::Value *emitJsonFree(CodeGen &cg, const CallExpr &e) {
+    cg.requireArgs(e, 1);
+    llvm::Value *val = cg.emitExpr(*e.args[0]);
+    if (rk_json_value < 0 || static_cast<size_t>(rk_json_value) >= cg.resource_sets_.size() ||
+        !lookupJsonSet(cg.resource_sets_[rk_json_value], val))
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)